src/components/forms/TagInput.tsx

```diff
@@ -0,0 +1,174 @@
+                        placeholder="Add tags"
+                        onInput={(event) => {
+                            setTags(
+                                (tags) =>
+                                    new Set(tags.add(event.currentTarget.value))
+                            )
+                            event.currentTarget.value = 'Select one...'
+                        }}
+                        value="Select one..."
+                    >
+                        <option disabled>Select one...</option>
+                        {options
+                            .filter((o) => !tags.has(o))
+                            .map((option, index) => (
+                                <option key={index}>{option}</option>
+                            ))}
+                    </select>
+                    <span class="absolute right-0 select-none" aria-hidden>
+                        <svg
+                            xmlns="http://www.w3.org/2000/svg"
+                            width="16"
+                            height="16"
+                            viewBox="0 0 24 24"
+                            fill="none"
+                            stroke="currentColor"
+                            stroke-width="2"
+                            stroke-linecap="round"
+                            stroke-linejoin="round"
+                        >
+                            <path d="m7 15 5 5 5-5"></path>
+                            <path d="m7 9 5-5 5 5"></path>
+                        </svg>
+                    </span>
+                </div>
+            </div>
+        </div>
+    )
+}
+
+export function TagInput(props: JSX.IntrinsicElements['input']) {
+    const [tags, setTags] = useState(new Set<string>())
+    const hydrated = useHydrated()
+
+    if (!hydrated) {
+        return (
+            <input
+                {...props}
+                class={inputClass}
+                placeholder="Add tags (separated by commas)"
+            />
+        )
+    }
+
+    return (
+        <div class={inputBaseClass}>
+            <input {...props} type="hidden" value={[...tags].join(', ')} />
+            <div class="flex flex-wrap items-center gap-2">
+                {/* extra wrapper element to make sure the input keeps its spot in the DOM and doesn't lose focus on tag updates */}
+                <div class="contents">
+                    <TagList
+                        tags={tags}
+                        onRemove={(tag) => {
+                            setTags((tags) => {
+                                tags.delete(tag)
+                                return new Set(tags)
+                            })
+                        }}
+                    />
+                </div>
+                <input
+                    class="flex-1 outline-none bg-transparent"
+                    placeholder="Add tags (separated by commas or spaces)"
+                    onKeyDown={(event) => {
```

Comment:
This should probably also run for onblur as well, tabbing out of the input box without hitting enter first doesn't add the keyword